### ISSUE TITLE
Error when compiling in Windows

### ISSUE THREAD

**Issue by loumalouomega** [beltoforion/muparser#90]
I am trying to use muparser in our project. I added in the following PR:

https://github.com/KratosMultiphysics/Kratos/pull/7899

Here the files are hardcopied (with the corresponding changes because missing overrides), see https://github.com/beltoforion/muparser/pull/89

We are doing a static compilation with the following CMakeLists.txt:

~~~cmake
SET(CMAKE_INCLUDE_CURRENT_DIR ON)

# Generate variables with the sources
FILE(GLOB_RECURSE MUPARSER_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/muparser/*.cpp") #all .cpp

# Compile library
ADD_LIBRARY(muparser STATIC ${MUPARSER_SOURCES})

# In case of using Cotire
IF(USE_COTIRE MATCHES ON)
    cotire(muparser)
ENDIF(USE_COTIRE MATCHES ON)

# Define include and library
SET(MUPARSER_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/muparser/)
SET(MUPARSER_LIBRARIES muparser)

# Static libraries
TARGET_COMPILE_DEFINITIONS(muparser PUBLIC MUPARSER_STATIC)

# Enable OpenMP
IF (KRATOS_SHARED_MEMORY_PARALLELIZATION STREQUAL "OpenMP")
    TARGET_COMPILE_DEFINITIONS(muparser PRIVATE MUP_USE_OPENMP)
ENDIF (KRATOS_SHARED_MEMORY_PARALLELIZATION STREQUAL "OpenMP")

# Printing a message
MESSAGE(STATUS "muparser : ${MUPARSER_INCLUDE_DIR}" )
~~~

The compilation fails in Windows as follows: (https://github.com/KratosMultiphysics/Kratos/pull/7899/checks?check_run_id=1577371304)

~~~sh
 D:\a\Kratos\Kratos\external_libraries\muparser\muparser\muParserBase.cpp(62,37): error C2491: 'mu::ParserBase::s_locale': definition of dllimport static data member not allowed (compiling source file D:\a\Kratos\Kratos\build\Custom\external_libraries\muparser\Custom\cotire\muparser_CXX_unity_0_4.cxx) [D:\a\Kratos\Kratos\build\Custom\external_libraries\muparser\muparser_unity.vcxproj]
...
~~~

Basically error C2491. Any idea how to solve it?

**Comment by loumalouomega** [comment]
I fix it with a manual definition:

~~~sh
ADD_DEFINITIONS( -DMUPARSERLIB_EXPORTS )
~~~

BTW, at the end I did a shared library instead of static

**Comment by beltoforion** [comment]
I have never seen issues when compiling with MSVC so i can't really say much about this.

**Comment by loumalouomega** [comment]
Strange indeed, but I found the solution :D

**Comment by wjheitler** [comment]
I have the same problem when compiling muParserBase.cpp. I'm upgrading a project that was using an earlier version of muParser (c 2013, not sure which version), and that project does not have this problem, but it has appeared when I change the mu files to the equivalent in v 2.3.2.
Could you tell me where you added your fix ADD_DEF...? Sorry to be dim, but was it somewhere in the MSVC project definition, or to one of the cpp/h files? Thanks for any help.
